packages/aws-cdk-lib/aws-logs/lib/pattern.ts

```
@@ -116,6 +116,27 @@ export class FilterPattern {
     return new JSONNumberPattern(jsonField, comparison, value);
   }
 
+  /**
+   * A JSON log pattern that compares against a Regex values.
+   *
+   * This pattern only matches if the event is a JSON event, and the indicated field inside
+   * compares with the regex value.
+   *
+   * Use '$' to indicate the root of the JSON structure. The comparison operator can only
+   * compare equality or inequality.
+   *
+   * For more information, see:
+   *
+   * https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/FilterAndPatternSyntax.html
+   *
+   * @param jsonField Field inside JSON. Example: "$.myField"
+   * @param comparison Comparison to carry out. Either = or !=.
+   * @param value The regex value to compare to.
+   */
+  public static regexValue(jsonField: string, comparison: '=' | '!=', value: string): JsonPattern {
```

Due to JSII language translation limitations, we're not able to support a union for the `comparison` type. As a workaround, you could provide two methods for each case: `regexValueEquals()` and `regexValueNotEquals()`.

---

I would keep the scope for now small, I think getting this in already unblocks a lot of use-cases.

---

We can throw an error if the comparison isn't `=` or `!=`. Better to throw error during synth time rather than the deployment failing